diff --git a/debian/changelog b/debian/changelog
index 0184181f..5f8a979d 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+pve-manager (8.3.0+wsh1) bookworm; urgency=medium
+
+  * Add remote audio and display support
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Sun, 24 Nov 2024 00:07:57 +0000
+
 pve-manager (8.3.0) bookworm; urgency=medium
 
   * ceph: drop preview flag for the Ceph Squid release
 
   * aplinfo: update TurnkeyLinux key after expiry
 
   * update shipped aplliance info index
 
   * ui: firewall rules: add empty text to grid
 
