diff --git a/debian/changelog b/debian/changelog
index d84b462..04a4afe 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,17 @@
+pve-qemu-kvm (9.1.2-3+wsh1) bookworm; urgency=medium
+
+  * Feat: Add extra binaries to pve qemu and reworked the pipeline
+    process for local testing via makefile targets (#4)
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Tue, 04 Feb 2025 20:41:47 +0000
+
 pve-qemu-kvm (9.1.2-3) bookworm; urgency=medium
 
   * async snapshot: explicitly specify raw format when loading the VM state
     file
 
   * vma create: rework CLI parameters for passing disk to a more structured
     style and use that to allow explicitly specifying the format
 
  -- Proxmox Support Team <support@proxmox.com>  Fri, 24 Jan 2025 16:12:34 +0100
 
