diff --git a/debian/changelog b/debian/changelog
index 5ef94ac..0b3ae90 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+pve-qemu-kvm (9.2.0-4+wsh1) bookworm; urgency=medium
+
+  * Add machine values and fix parsing in UI
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Fri, 04 Apr 2025 20:50:49 +0000
+
 pve-qemu-kvm (9.2.0-4) bookworm; urgency=medium
 
   * various async snapshot improvements, inclduing using a dedicated IO thread
     for the state file when doing a live snapshot. That should reduce load on
     the main thread and for it to get stuck on IO, i.e. same benefits as using
     a dedicated IO thread for regular drives. This is particularly interesting
     when the VM state storage is a network storage like NFS. It should also
     address #6262.
 
   * pve backup: implement basic features and API in preperation for external
