diff --git a/debian/changelog b/debian/changelog
index d04a78c6..503dcdf3 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,22 @@
+pve-manager (8.3.5+wsh2) bookworm; urgency=medium
+
+  * chore: Update patches to reset offset
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Sun, 13 Apr 2025 20:21:05 +0000
+
+pve-manager (8.3.5+wsh1) bookworm; urgency=medium
+
+  * chore: Update patches to reset offset
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Fri, 11 Apr 2025 22:45:48 +0000
+
 pve-manager (8.3.5) bookworm; urgency=medium
 
   * update shipped appliance info index
 
   * fix #6138: ui: storage content: use locale for sorting
 
   * ui: sdn ipam: powerDNS, NetBox, phpIPAM: split layout in two columns
 
   * ui: sdn ipam: powerDNS, NetBox, phpIPAM: add fingerprint textfield
 
