diff --git a/debian/changelog b/debian/changelog
index c47e7cf5..a6bfb876 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+pve-manager (9.1.10+wsh1) trixie; urgency=medium
+
+  * Refresh patches: pve-manager,qemu-server
+
+ -- hwinther <hwinther@users.noreply.github.com>  Tue, 12 May 2026 20:46:41 +0000
+
 pve-manager (9.1.10) trixie; urgency=medium
 
   * ui: access control: tokens: add a button to regenerate the secret of
     an existing API token.
 
   * ui: qemu: also use 'x86-64-v2-AES' as default in the dedicated
     processor edit panel, matching what the create wizard already
     suggested, so it is harder to accidentally end up on the discouraged
     backend default 'kvm64' by clearing the field.
 
