diff --git a/debian/changelog b/debian/changelog
index 35281091..ed6780f3 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+pve-manager (9.1.4+wsh1) trixie; urgency=medium
+
+  * feat: proxmox 9 patches (#56)
+
+ -- Hans Christian Winther-Sorensen <hc@wsh.no>  Sat, 27 Dec 2025 20:39:45 +0000
+
 pve-manager (9.1.4) trixie; urgency=medium
 
   * api: apt changelog: check for odd package name and ensure arguments are
     separated from options.
 
  -- Proxmox Support Team <support@proxmox.com>  Mon, 22 Dec 2025 12:02:58 +0100
 
 pve-manager (9.1.3) trixie; urgency=medium
 
   * ui: ceph install wizard: fix typo.
