diff --git a/debian/changelog b/debian/changelog
index 8fe0bdb7..5ba79fba 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+pve-manager (9.2.6+wsh1) trixie; urgency=medium
+
+  * Refresh patches: pve-manager
+
+ -- hwinther <hwinther@users.noreply.github.com>  Sat, 01 Aug 2026 09:59:27 +0000
+
 pve-manager (9.2.6) trixie; urgency=medium
 
   * fabrics: wireguard: node edit: use network selector for internal nodes,
     making the semantics of the field more obvious, and entering values less
     error-prone and manual.
 
   * api: apt: versions: track proxmox-enterprise-support-keyring version
 
   * ui: ha: node affinity: refactor and extend UI, allow configuring "positive"
     or "negative" affinity rules. negative affinity allows declaring nodes on
