diff --git a/Makefile b/Makefile
index f23b041..cad0c22 100644
--- a/Makefile
+++ b/Makefile
@@ -26,8 +26,6 @@ PC_BIOS_FW_PURGE_LIST_IN = \
 	hppa-firmware.img \
 	hppa-firmware64.img \
 	openbios-ppc \
-	openbios-sparc32 \
-	openbios-sparc64 \
 	palcode-clipper \
 	s390-ccw.img \
 	s390-netboot.img \
diff --git a/debian/pve-qemu-kvm.lintian-overrides b/debian/pve-qemu-kvm.lintian-overrides
index 6fb5116..799c744 100644
--- a/debian/pve-qemu-kvm.lintian-overrides
+++ b/debian/pve-qemu-kvm.lintian-overrides
@@ -1,7 +1,13 @@
 pve-qemu-kvm: arch-dependent-file-in-usr-share [usr/share/kvm/hppa-firmware.img]
+pve-qemu-kvm: arch-dependent-file-in-usr-share [usr/share/kvm/openbios-sparc32]
+pve-qemu-kvm: arch-dependent-file-in-usr-share [usr/share/kvm/openbios-sparc64]
 pve-qemu-kvm: binary-from-other-architecture [usr/share/kvm/hppa-firmware.img]
+pve-qemu-kvm: binary-from-other-architecture [usr/share/kvm/openbios-sparc32]
+pve-qemu-kvm: binary-from-other-architecture [usr/share/kvm/openbios-sparc64]
 pve-qemu-kvm: embedded-javascript-library please use * [usr/share/doc/pve-qemu-kvm/kvm/_static/*]
 pve-qemu-kvm: groff-message *: warning [*]: can't break line [usr/share/man/*]
 pve-qemu-kvm: groff-message *: warning [*]: cannot adjust line [usr/share/man/*]
 pve-qemu-kvm: statically-linked-binary [usr/share/kvm/hppa-firmware.img]
+pve-qemu-kvm: statically-linked-binary [usr/share/kvm/openbios-sparc32]
+pve-qemu-kvm: statically-linked-binary [usr/share/kvm/openbios-sparc64]
 pve-qemu-kvm: unstripped-binary-or-object [usr/share/kvm/hppa-firmware.img]
diff --git a/debian/rules b/debian/rules
index 51f56c5..2532d30 100755
--- a/debian/rules
+++ b/debian/rules
@@ -18,7 +18,7 @@ machinefile := $(destdir)/usr/share/kvm/machine-versions-x86_64.json
 BUILDDIR=build
 
 # FIXME: pass to configure as --extra-cflags=CFLAGS ?! also LDFLAGS?
-CFLAGS += -Wall
+CFLAGS += -Wall -mcrc32  # crc32 added for qemu-3dfx mesa hw
 
 export CFLAGS
 
@@ -43,19 +43,19 @@ endif
 	    --localstatedir=/var \
 	    --prefix=/usr \
 	    --sysconfdir=/etc \
-	    --target-list=$(HOST_CPU)-softmmu,aarch64-softmmu \
+	    --target-list=$(HOST_CPU)-softmmu,aarch64-softmmu,sparc-softmmu,sparc64-softmmu \
 	    --with-suffix="kvm" \
 	    --with-pkgversion="${DEB_SOURCE}_${DEB_VERSION_UPSTREAM_REVISION}" \
-	    --audio-drv-list="alsa" \
+	    --audio-drv-list="alsa,pa,pipewire" \
 	    --datadir=/usr/share \
 	    --libexecdir=/usr/lib/kvm \
 	    --disable-capstone \
-	    --disable-gtk \
+	    --enable-gtk \
 	    --disable-guest-agent \
 	    --disable-guest-agent-msi \
 	    --disable-libnfs \
 	    --disable-libssh \
-	    --disable-sdl \
+	    --enable-sdl \
 	    --disable-smartcard \
 	    --disable-strip \
 	    --disable-xen \
@@ -119,6 +119,32 @@ install: build
 	rm -f $(destdir)/usr/lib/kvm/qemu-bridge-helper
 	rm -f $(destdir)/usr/lib/kvm/virtfs-proxy-helper
 
+# Add extra binaries, sparc backport and 3dfx passthrough
+ifneq ("$(wildcard ../../../build/pve-qemu-7.2-sparc/qemu-system-sparc)", "")
+	cp ../../../build/pve-qemu-7.2-sparc/qemu-system-sparc $(destdir)/usr/bin/qemu-system-sparc-7.2
+endif
+
+ifneq ("$(wildcard ../../../build/pve-qemu-7.2-sparc/qemu-system-sparc64)", "")
+	cp ../../../build/pve-qemu-7.2-sparc/qemu-system-sparc64 $(destdir)/usr/bin/qemu-system-sparc64-7.2
+endif
+
+ifneq ("$(wildcard ../../../build/pve-qemu-3dfx/qemu-system-x86_64)", "")
+	cp ../../../build/pve-qemu-3dfx/qemu-system-x86_64 $(destdir)/usr/bin/qemu-system-x86_64-3dfx
+endif
+
+# Add extra system-sparc bios images
+ifneq ("$(wildcard ../../sparc/ss5.bin)", "")
+	cp ../../sparc/ss5.bin $(destdir)/usr/share/kvm/
+endif
+
+ifneq ("$(wildcard ../../sparc/ss10_v2.25_rom)", "")
+	cp ../../sparc/ss10_v2.25_rom $(destdir)/usr/share/kvm/
+endif
+
+ifneq ("$(wildcard ../../sparc/ss20_v2.25_rom)", "")
+	cp ../../sparc/ss20_v2.25_rom $(destdir)/usr/share/kvm/
+endif
+
 	# CPU flags are static for QEMU version, allows avoiding more costly checks
 	$(destdir)/usr/bin/qemu-system-x86_64 -cpu help | ./debian/parse-cpu-flags.pl > $(flagfile)
 	$(destdir)/usr/bin/qemu-system-x86_64 -machine help | ./debian/parse-machines.pl > $(machinefile)
@@ -148,7 +174,7 @@ binary-arch: build install
 #	dh_installinfo
 	dh_installman
 	dh_link
-	dh_strip --dbgsym-migration='pve-qemu-kvm-dbg (<<8.0.0-1~)'
+	dh_strip --dbgsym-migration='pve-qemu-kvm-dbg (<<8.0.0-1~)' --exclude="openbios"
 	dh_compress
 	dh_fixperms
 #	dh_perl
