diff --git a/debian/changelog b/debian/changelog
index 82b09641..ffb17256 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,22 @@
+pve-manager (8.3.4+wsh2) bookworm; urgency=medium
+
+  * Move helper qemu server helper and reset line drift
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Thu, 06 Mar 2025 20:21:20 +0000
+
+pve-manager (8.3.4+wsh1) bookworm; urgency=medium
+
+  * Move helper qemu server helper and reset line drift
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Sat, 22 Feb 2025 20:59:03 +0000
+
 pve-manager (8.3.4) bookworm; urgency=medium
 
   * fix #5699: add support to define a HTTP header from which the real IP of a
     connection should be parsed from. This can be useful for setups with a
     reverse proxy in front of the API server. On top of that add support for
     optionally configuring an allow-list of IP networks that the real source
     IP must match one to allow the connection to be handled.
     The related config variables are named `PROXY_REAL_IP_HEADER` and
     `PROXY_REAL_IP_ALLOW_FROM` respectively.
 
