diff --git a/debian/changelog b/debian/changelog
index 713b3e4..1723236 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+pve-qemu-kvm (11.1.1-1+wsh1) trixie; urgency=medium
+
+  * Refresh patches: qemu-server,pve-qemu
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Thu, 01 Oct 2026 00:00:51 +0000
+
 pve-qemu-kvm (11.1.1-1) trixie; urgency=medium
 
   * update submodule and patches to QEMU 11.1.1.
 
   * the glusterfs block driver is fully dropped from this QEMU version. The
     Proxmox VE storage plugin was already removed with the release of Proxmox
     VE 9.0.
 
   * update supported CPU models for amd64/x86_64 adding new versions for some
     AMD and Intel models, which add GMET/MBEC support.
