diff --git a/debian/changelog b/debian/changelog
index 5bcbd764..77e67fc4 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+pve-manager (8.2.6+wsh1) bookworm; urgency=medium
+
+  * Update pve-manager patch
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Sun, 22 Sep 2024 21:52:38 +0000
+
 pve-manager (8.2.6) bookworm; urgency=medium
 
   * fix #5731: vzdump jobs: fix execution of converted jobs
 
  -- Proxmox Support Team <support@proxmox.com>  Fri, 20 Sep 2024 17:47:17 +0200
 
 pve-manager (8.2.5) bookworm; urgency=medium
 
   * ui: qemu hardware: use background delay for asynchronous remove tasks so
     that the task log window doesn't get shown if the task already finished.
