diff --git a/debian/changelog b/debian/changelog
index abc433e4..a424432e 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+pve-manager (9.2.10+wsh1) trixie; urgency=medium
+
+  * Refresh patches: pve-manager
+
+ -- hwinther <hwinther@users.noreply.github.com>  Mon, 17 Aug 2026 20:40:11 +0000
+
 pve-manager (9.2.10) trixie; urgency=medium
 
   * pveperf: make bogomips regular expression case insensitive, other
     architectures like Loong64 use different capitalization
 
   * ui: LXC templates: hide foreign architectures by default, `arm64` and
     `amd64` look very similar and can be easily confused
 
   * ui: appliance templates: rename the "Package" column to "Name"
 
