diff --git a/debian/changelog b/debian/changelog
index 46371ae..952e90a 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,22 @@
+pve-qemu-kvm (9.0.2-2+wsh2) bookworm; urgency=medium
+
+  * Remove unused lintian override
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Sun, 18 Aug 2024 21:34:54 +0000
+
+pve-qemu-kvm (9.0.2-2+wsh1) bookworm; urgency=medium
+
+  * Add pve-qemu WSH patches
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Sun, 18 Aug 2024 21:12:59 +0000
+
 pve-qemu-kvm (9.0.2-2) bookworm; urgency=medium
 
   * actually update submodule to QEMU 9.0.2. The previous release was still
     based on 9.0.0 by mistake.
 
  -- Proxmox Support Team <support@proxmox.com>  Wed, 07 Aug 2024 10:16:01 +0200
 
 pve-qemu-kvm (9.0.2-1) bookworm; urgency=medium
 
   * update submodule and patches to QEMU 9.0.2. While our version had most
