diff --git a/debian/changelog b/debian/changelog
index 5045164..7940689 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+pve-qemu-kvm (10.1.2-3+wsh1) trixie; urgency=medium
+
+  * feat: proxmox 9 patches (#56)
+
+ -- Hans Christian Winther-Sorensen <hc@wsh.no>  Tue, 18 Nov 2025 16:55:33 +0000
+
 pve-qemu-kvm (10.1.2-3) trixie; urgency=medium
 
   * cherry-pick fix for a regression with VNC clipboard breaking mouse pointer
     in Windows guests.
 
  -- Proxmox Support Team <support@proxmox.com>  Mon, 17 Nov 2025 16:42:03 +0100
 
 pve-qemu-kvm (10.1.2-2) trixie; urgency=medium
 
   * enable FUSE support in preparation to allow FUSE exports of, e.g.,
