diff --git a/debian/changelog b/debian/changelog
index 304d4243..a2789ab1 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+pve-manager (9.1.9+wsh1) trixie; urgency=medium
+
+  * Refresh patches: pve-manager,qemu-server
+
+ -- hwinther <hwinther@users.noreply.github.com>  Thu, 23 Apr 2026 08:41:24 +0000
+
 pve-manager (9.1.9) trixie; urgency=medium
 
   * ui: ha: status: handle the 'fencing' entry when ordering the grid, so it
     no longer bounces around between API requests on some browsers. Place it
     directly after 'quorum' for quick visibility.
 
   * api: apt package versions: fix typo in libpve-notify-perl package name,
     which caused the package to be missing from the versions list.
 
   * fix #6234: ui: ceph: make the CephFS page scrollable along the Y-axis, so
