diff --git a/debian/changelog b/debian/changelog
index d04a78c6..a33f8b55 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,28 @@
+pve-manager (8.3.5+wsh3) bookworm; urgency=medium
+
+  * Improve handling of machine values for sun
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Sun, 16 Mar 2025 20:58:52 +0000
+
+pve-manager (8.3.5+wsh2) bookworm; urgency=medium
+
+  * Add machine values and fix parsing in UI
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Sat, 15 Mar 2025 17:58:57 +0000
+
+pve-manager (8.3.5+wsh1) bookworm; urgency=medium
+
+  * Add custom color theme
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Sat, 08 Mar 2025 22:51:25 +0000
+
 pve-manager (8.3.5) bookworm; urgency=medium
 
   * update shipped appliance info index
 
   * fix #6138: ui: storage content: use locale for sorting
 
   * ui: sdn ipam: powerDNS, NetBox, phpIPAM: split layout in two columns
 
   * ui: sdn ipam: powerDNS, NetBox, phpIPAM: add fingerprint textfield
 
