diff --git a/debian/changelog b/debian/changelog
index f83e130..83862b3 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+pve-qemu-kvm (9.2.0-5+wsh1) bookworm; urgency=medium
+
+  * chore: Update patches to reset offset
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Fri, 11 Apr 2025 18:52:56 +0000
+
 pve-qemu-kvm (9.2.0-5) bookworm; urgency=medium
 
   * pve backup: backup-access api: simplify bitmap logic
 
  -- Proxmox Support Team <support@proxmox.com>  Fri, 04 Apr 2025 16:15:58 +0200
 
 pve-qemu-kvm (9.2.0-4) bookworm; urgency=medium
 
   * various async snapshot improvements, inclduing using a dedicated IO thread
     for the state file when doing a live snapshot. That should reduce load on
