diff --git a/debian/changelog b/debian/changelog
index 37c67de1..16cb2e91 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,22 @@
+pve-manager (9.2.11+wsh2) trixie; urgency=medium
+
+  * Refresh patches: pve-manager
+
+ -- hwinther <hwinther@users.noreply.github.com>  Wed, 26 Aug 2026 23:26:40 +0000
+
+pve-manager (9.2.11+wsh1) trixie; urgency=medium
+
+  * Refresh patches: pve-manager
+
+ -- hwinther <hwinther@users.noreply.github.com>  Tue, 18 Aug 2026 08:36:14 +0000
+
 pve-manager (9.2.11) trixie; urgency=medium
 
   * ui: certificates, lxc DNS, VM cloud-init and PCI: encode rendered values
     with htmlEncode.
 
   * ui: acme: use improved URL rendering implementation from
     proxmox-widget-toolkit.
 
  -- Proxmox Support Team <support@proxmox.com>  Fri, 14 Aug 2026 14:27:06 +0200
 
