diff --git a/debian/changelog b/debian/changelog
index fadc85b..bec85ab 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,17 @@
+pve-qemu-kvm (9.1.2-1+wsh1) bookworm; urgency=medium
+
+  * Feat: Add extra binaries to pve qemu and reworked the pipeline
+    process for local testing via makefile targets (#4)
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Fri, 24 Jan 2025 20:41:31 +0000
+
 pve-qemu-kvm (9.1.2-1) bookworm; urgency=medium
 
   * update submodule and patches to QEMU 9.1.2
 
   * improve error handling and edge cases with fleecing backups.
 
  -- Proxmox Support Team <support@proxmox.com>  Wed, 11 Dec 2024 16:47:21 +0100
 
 pve-qemu-kvm (9.0.2-4) bookworm; urgency=medium
 
