diff --git a/debian/changelog b/debian/changelog
index 916a356f..fd22ae63 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,22 @@
+pve-manager (8.2.7+wsh2) bookworm; urgency=medium
+
+  * Add remote audio and display support
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Fri, 27 Sep 2024 21:55:14 +0000
+
+pve-manager (8.2.7+wsh1) bookworm; urgency=medium
+
+  * Add remote audio and display support
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Tue, 24 Sep 2024 18:52:33 +0000
+
 pve-manager (8.2.7) bookworm; urgency=medium
 
   * api: add missing schema annotations for download endpoints and adapt
     existing ones to renaming the respective property to 'download_allowed'.
 
   * pvesh: limit download handling to annotated endpoints to unify the
     behaviour with pve-http-server.
 
  -- Proxmox Support Team <support@proxmox.com>  Mon, 23 Sep 2024 11:43:24 +0200
 
