diff --git a/debian/changelog b/debian/changelog
index 6a09f4a2..3b098422 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,40 @@
+pve-manager (8.4.1+wsh5) bookworm; urgency=medium
+
+  * chore: Update patches to reset offset
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Fri, 11 Apr 2025 22:25:40 +0000
+
+pve-manager (8.4.1+wsh4) bookworm; urgency=medium
+
+  * chore: Update patches to reset offset
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Fri, 11 Apr 2025 22:22:19 +0000
+
+pve-manager (8.4.1+wsh3) bookworm; urgency=medium
+
+  * chore: Update patches to reset offset
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Fri, 11 Apr 2025 22:16:58 +0000
+
+pve-manager (8.4.1+wsh2) bookworm; urgency=medium
+
+  * chore: Update patches to reset offset
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Fri, 11 Apr 2025 18:23:01 +0000
+
+pve-manager (8.4.1+wsh1) bookworm; urgency=medium
+
+  * Improve handling of machine values for sun
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Thu, 10 Apr 2025 21:27:11 +0000
+
 pve-manager (8.4.1) bookworm; urgency=medium
 
   * ui: fix regression with checking if volume is QEMU backup, which could
     block some dialogues like file-restore.
 
   * ceph versions: do not try to parse empty string and include raw string in
     error to make this easier to debug.
 
   * ui: backup: hide change mode detection for VMs and non-PBS storage.
 
