diff --git a/debian/changelog b/debian/changelog
index 8c2507a..f43fc4f 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+pve-qemu-kvm (10.1.2-1+wsh1) bookworm; urgency=medium
+
+  * fix: Add aarch64 edk2 firmware as a runtime dependency
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Fri, 14 Nov 2025 20:41:14 +0000
+
 pve-qemu-kvm (10.1.2-1) trixie; urgency=medium
 
   * fix #6810: fix a potential deadlock upon TMF request cancelling with
     VirtIO SCSI
 
   * update QEMU and downstream patches for 10.1.2 stable release.
 
   * cherry pick fix for VFIO regression affecting v10.1, which, e.g., could
     affect some GPU passthrough setups.
 
