diff --git a/debian/changelog b/debian/changelog
index c1183865..1107a9fe 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,22 @@
+pve-manager (9.2.12+wsh2) trixie; urgency=medium
+
+  * Refresh patches: pve-manager
+
+ -- hwinther <hwinther@users.noreply.github.com>  Sat, 29 Aug 2026 02:48:31 +0000
+
+pve-manager (9.2.12+wsh1) trixie; urgency=medium
+
+  * Refresh patches: pve-manager
+
+ -- hwinther <hwinther@users.noreply.github.com>  Thu, 27 Aug 2026 18:53:16 +0000
+
 pve-manager (9.2.12) trixie; urgency=medium
 
   * fix #7136: ui: tree: harmonize folder view resource ordering
 
   * fix #6113: ui: import: allow selecting a resource pool when importing a
     guest from an OVA file or from VMware ESXi
 
   * ceph: rolling restart:
     - allow a restart on a cluster whose only errors are the cephx cipher
       health checks that Ceph 19.2.6 and 20.2.4 raise. Those made every
