diff --git a/debian/changelog b/debian/changelog
index a5426d9f..2cd0f406 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+pve-manager (8.3.1+wsh1) bookworm; urgency=medium
+
+  * Add 3dfx/mesa guest configuration
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Mon, 16 Dec 2024 20:21:41 +0000
+
 pve-manager (8.3.1) bookworm; urgency=medium
 
   * ui: sdn firewall: add online help reference
 
   * ui: cluster join info: add clipboard icon to "Copy Information" button
 
   * external metric server: influxdb: always quote tags
 
   * api: notification targets: fix permission check for POST/PUT
 
