diff --git a/debian/changelog b/debian/changelog
index fcbe39b..1bd1f06 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+pve-qemu-kvm (10.0.2-2+wsh1) bookworm; urgency=medium
+
+  * fix: Add aarch64 edk2 firmware as a runtime dependency
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Thu, 03 Jul 2025 20:41:49 +0000
+
 pve-qemu-kvm (10.0.2-2) trixie; urgency=medium
 
   * drop support for native Gl
   * drop support for accessing Gluster based storage directly due to its
     effective end of support and maintenance. The last upstream release
     happened over 2.5 years ago and there's currently no one providing
     enterprise support or security updates. Further, upstream QEMU will remove
     the integration in one of the next releases, so use the upcomming PVE 9
     major release to provide a clean cut.
     User can either stay on Proxmox VE 8 until its end-of-life (probably end
