diff --git a/debian/changelog b/debian/changelog
index 545ec76e..719d8719 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+pve-manager (8.3.3+wsh1) bookworm; urgency=medium
+
+  * Add 3dfx/mesa guest configuration
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Sat, 01 Feb 2025 22:08:38 +0000
+
 pve-manager (8.3.3) bookworm; urgency=medium
 
   * update shipped appliance info index
 
   * add missing htmlEncode for some UI elements
 
  -- Proxmox Support Team <support@proxmox.com>  Mon, 20 Jan 2025 11:37:32 +0100
 
 pve-manager (8.3.2) bookworm; urgency=medium
 
