diff --git a/debian/changelog b/debian/changelog
index 9cf815c..853ff5a 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+pve-qemu-kvm (9.0.2-3+wsh1) bookworm; urgency=medium
+
+  * Add experimental audio support
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Wed, 11 Sep 2024 21:42:53 +0000
+
 pve-qemu-kvm (9.0.2-3) bookworm; urgency=medium
 
   * pick up fix for VirtIO PCI regressions
 
   * pick up stable fixes for 9.0, including fixes for VirtIO-net, ARM and
     x86(_64) emulation, CVEs to harden NBD server against malicious clients,
     as well as a few others (VNC, physmem, Intel IOMMU, ...).
 
  -- Proxmox Support Team <support@proxmox.com>  Fri, 06 Sep 2024 16:21:42 +0200
 
