diff --git a/debian/changelog b/debian/changelog
index 271a834..2d43d4c 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+pve-qemu-kvm (10.2.1-1+wsh1) trixie; urgency=medium
+
+  * Manual fix needed: deb build failed (patch trigger) (#75)
+
+ -- Hans Christian Winther-Sorensen <hc@wsh.no>  Mon, 04 May 2026 20:54:43 +0000
+
 pve-qemu-kvm (10.2.1-1) trixie; urgency=medium
 
   * update QEMU and downstream patches for 10.2.1 release.
 
   * cherry-pick various stable fixes to improve 10.2 experience.
 
  -- Proxmox Support Team <support@proxmox.com>  Mon, 16 Mar 2026 23:05:33 +0100
 
 pve-qemu-kvm (10.1.2-7) trixie; urgency=medium
 
