diff --git a/debian/changelog b/debian/changelog
index 2c012e3..9ebecf1 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+pve-qemu-kvm (10.0.2-3+wsh1) bookworm; urgency=medium
+
+  * fix: Add aarch64 edk2 firmware as a runtime dependency
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Tue, 15 Jul 2025 20:42:33 +0000
+
 pve-qemu-kvm (10.0.2-3) trixie; urgency=medium
 
   * add backup/zeroinit/track-alloc blockdev patches
 
  -- Proxmox Support Team <support@proxmox.com>  Thu, 03 Jul 2025 12:02:03 +0200
 
 pve-qemu-kvm (10.0.2-2) trixie; urgency=medium
 
   * drop support for native Gl
   * drop support for accessing Gluster based storage directly due to its
