diff --git a/debian/changelog b/debian/changelog
index 7d5b367..9fae9b1 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,22 @@
+pve-qemu-kvm (10.1.2-2+wsh2) trixie; urgency=medium
+
+  * feat: proxmox 9 patches (#56)
+
+ -- Hans Christian Winther-Sorensen <hc@wsh.no>  Tue, 18 Nov 2025 08:11:01 +0000
+
+pve-qemu-kvm (10.1.2-2+wsh1) trixie; urgency=medium
+
+  * feat: proxmox 9 patches (#56)
+
+ -- Hans Christian Winther-Sorensen <hc@wsh.no>  Tue, 18 Nov 2025 00:09:15 +0000
+
 pve-qemu-kvm (10.1.2-2) trixie; urgency=medium
 
   * enable FUSE support in preparation to allow FUSE exports of, e.g.,
     qcow2-formatted TPM state volumes via qemu-storage-daemon.
 
  -- Proxmox Support Team <support@proxmox.com>  Thu, 13 Nov 2025 23:55:54 +0100
 
 pve-qemu-kvm (10.1.2-1) trixie; urgency=medium
 
   * fix #6810: fix a potential deadlock upon TMF request cancelling with
