diff --git a/debian/changelog b/debian/changelog
index 6730313..6ec7119 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+pve-qemu-kvm (10.1.2-6+wsh1) trixie; urgency=medium
+
+  * feat: proxmox 9 patches (#56)
+
+ -- Hans Christian Winther-Sorensen <hc@wsh.no>  Tue, 17 Feb 2026 20:24:55 +0000
+
 pve-qemu-kvm (10.1.2-6) trixie; urgency=medium
 
   * fix /usr/bin/kvm symlink to point to actual host architecture.
 
   * generate list of supported aarch64 machines during build and ship as JSON
     alongside the existing list for x86-64/amd64 machines in the resulting
     package.
 
  -- Proxmox Support Team <support@proxmox.com>  Wed, 04 Feb 2026 11:05:36 +0100
 
