diff --git a/debian/changelog b/debian/changelog
index ad28247d..e9f85d41 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+pve-manager (8.2.10+wsh1) bookworm; urgency=medium
+
+  * Add remote audio and display support
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Wed, 20 Nov 2024 20:21:33 +0000
+
 pve-manager (8.2.10) bookworm; urgency=medium
 
   * ui: firewall: add sdn as valid scope to IP-set selector
 
   * ui: firewall: add forward direction to rule panel
 
   * ui: sdn: add firewall panel
 
   * ui: firewall: show warning when creating forward rules
 
