diff --git a/debian/changelog b/debian/changelog
index 864895ab..d8d51a0c 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,22 @@
+pve-manager (8.2.8+wsh2) bookworm; urgency=medium
+
+  * Add remote audio and display support
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Tue, 19 Nov 2024 20:21:34 +0000
+
+pve-manager (8.2.8+wsh1) bookworm; urgency=medium
+
+  * Add remote audio and display support
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Mon, 11 Nov 2024 23:10:16 +0000
+
 pve-manager (8.2.8) bookworm; urgency=medium
 
   * fix #5734: provide missing methods for Proxmox.Utils for mobile UI
 
   * api: subscription update: report error if removing ceph-auth file fails
 
   * add module to centrally track Ceph release info for the backend
 
   * pveceph: prompt user on TTY about installing experimental version
 
