diff --git a/debian/changelog b/debian/changelog
index 8ec122b9..623dc640 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+qemu-server (9.1.16+wsh1) trixie; urgency=medium
+
+  * Refresh patches: qemu-server
+
+ -- hwinther <hwinther@users.noreply.github.com>  Wed, 17 Jun 2026 20:58:51 +0000
+
 qemu-server (9.1.16) trixie; urgency=medium
 
   * api schema: vm resume: add description for the nocheck parameter.
 
  -- Proxmox Support Team <support@proxmox.com>  Sun, 24 May 2026 02:01:31 +0200
 
 qemu-server (9.1.15) trixie; urgency=medium
 
   * cpu models: only include custom CPU models matching the host architecture
     in the model listing, instead of also showing custom models with an
