diff --git a/debian/changelog b/debian/changelog
index 4dc3d9f2..a6ba57a0 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+qemu-server (9.0.0+wsh1) bookworm; urgency=medium
+
+  * fix: Support for sun machine values
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Thu, 03 Jul 2025 20:21:36 +0000
+
 qemu-server (9.0.0) trixie; urgency=medium
 
   * re-build for Debian 13 Trixie based Proxmox VE 9 releases.
 
   * move module-load config from /etc to /usr, as a file with the same name in
     the /etc/modules-load.d/ directory takes precedence over a file in the
     /usr/lib/modules-load.d/ directory, an admin can still overrides this in a
     easy manner.
 
   * drop duplicated network hook script from /var, they have been migrated to
