diff --git a/debian/changelog b/debian/changelog
index d44ab06a..7834b568 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+qemu-server (9.2.1+wsh1) trixie; urgency=medium
+
+  * Refresh patches: qemu-server,pve-qemu
+
+ -- hwinther <hwinther@users.noreply.github.com>  Sun, 26 Jul 2026 09:59:25 +0000
+
 qemu-server (9.2.1) trixie; urgency=medium
 
   * fix #7828: register unused drive: fix failure to handle non-PVE managed
     volumes, such as pass-through host block devices
 
   * ensure modules.load snippet in /etc is correctly cleaned up on upgrades
 
  -- Proxmox Support Team <support@proxmox.com>  Fri, 24 Jul 2026 11:12:14 +0200
 
 qemu-server (9.2.0) trixie; urgency=medium
