diff --git a/debian/changelog b/debian/changelog
index c531e7d5..990e0637 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+qemu-server (8.3.12+wsh1) bookworm; urgency=medium
+
+  * Add error message when arch binary is missing
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Fri, 11 Apr 2025 18:13:13 +0000
+
 qemu-server (8.3.12) bookworm; urgency=medium
 
   * virtiofs: drop unsafe and problematic writeback option completely fow now.
 
  -- Proxmox Support Team <support@proxmox.com>  Tue, 08 Apr 2025 17:30:49 +0200
 
 qemu-server (8.3.11) bookworm; urgency=medium
 
   * handle various deprecations of options in upstream QEMU 9.2.
 
