diff --git a/debian/changelog b/debian/changelog
index d61940fe..ec44bddf 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+qemu-server (9.1.2+wsh1) trixie; urgency=medium
+
+  * feat: proxmox 9 patches (#56)
+
+ -- Hans Christian Winther-Sorensen <hc@wsh.no>  Wed, 17 Dec 2025 20:22:22 +0000
+
 qemu-server (9.1.2) trixie; urgency=medium
 
   * check efi vars: avoid warning about accessing an undefined value from Perl
     when no OS type is set.
 
   * query machine capabilities: remove left-over debug print.
 
   * dbus-vmstate: fix method call on dbus object resolving to wrong instance.
 
   * migrate: remove left-over dbus-vmstate instance when migrating without
