diff --git a/debian/changelog b/debian/changelog
index 7d440810..0d27209f 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+qemu-server (8.3.6+wsh1) bookworm; urgency=medium
+
+  * Add 3dfx/mesa guest configuration
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Sat, 01 Feb 2025 22:11:02 +0000
+
 qemu-server (8.3.6) bookworm; urgency=medium
 
   * api: do not set an empty machine option when pinning
 
  -- Proxmox Support Team <support@proxmox.com>  Mon, 20 Jan 2025 14:45:58 +0100
 
 qemu-server (8.3.5) bookworm; urgency=medium
 
   * unify CDROM medium hotplug and startup code
 
