diff --git a/debian/changelog b/debian/changelog
index 06743f6..50924f6 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+qemu-server (8.2.6+wsh1) bookworm; urgency=medium
+
+  * Add remote audio and display support
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Mon, 11 Nov 2024 22:31:49 +0000
+
 qemu-server (8.2.6) bookworm; urgency=medium
 
   * vm start: add syslog info with which PID a VM process was started
 
   * pci: avoid the hard requirement for a passthrough device to be reset on
     start to fix a recent regression from improving error handling from the
     pve-common package.
 
  -- Proxmox Support Team <support@proxmox.com>  Mon, 11 Nov 2024 20:38:25 +0100
 
