diff --git a/debian/changelog b/debian/changelog
index 49bda363..87fe3c3d 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,22 @@
+qemu-server (8.3.8+wsh2) bookworm; urgency=medium
+
+  * Fix variable comparison
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Sat, 22 Feb 2025 21:42:49 +0000
+
+qemu-server (8.3.8+wsh1) bookworm; urgency=medium
+
+  * Move helper qemu server helper and reset line drift
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Sat, 22 Feb 2025 20:59:03 +0000
+
 qemu-server (8.3.8) bookworm; urgency=medium
 
   * resume: error out if VM is a template
 
   * backup: also restore VM power state for early failures
 
   * config: add special class that prevents writing the configuration
 
   * fix #6007: template backup: use minimized configuration for handling the full vm start
 
