diff --git a/debian/changelog b/debian/changelog
index ba09e1fd..6f6dbc86 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+qemu-server (9.1.8+wsh1) trixie; urgency=medium
+
+  * Refresh patches: pve-manager,qemu-server
+
+ -- hwinther <hwinther@users.noreply.github.com>  Thu, 23 Apr 2026 08:41:24 +0000
+
 qemu-server (9.1.8) trixie; urgency=medium
 
   * cloud-init: dns: avoid "uninitialized value in split" warning if there is
     no 'searchdomain' cloud-init setting and also no 'search' setting in the
     host's resolv.conf.
 
   * api: dump cloud-init: mask password for user without VM.Config.Cloudinit
     privilege.
 
   * api: confirm Sys.Console privilege when creating or restoring a VM and
