diff --git a/debian/changelog b/debian/changelog
index ba9970a8..836e3264 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+qemu-server (9.2.6+wsh1) trixie; urgency=medium
+
+  * Refresh patches: qemu-server,pve-qemu
+
+ -- hwinther <hwinther@users.noreply.github.com>  Sun, 23 Aug 2026 20:39:10 +0000
+
 qemu-server (9.2.6) trixie; urgency=medium
 
   * cloud-init: commit: activate volume early enough to fix regression with
     qcow2 on LVM.
 
   * api: hostpci: restore permission check for 'romfile' option that was
     accidentally changed when PCI mappings were introduced.
 
  -- Proxmox Support Team <support@proxmox.com>  Mon, 17 Aug 2026 13:32:45 +0200
 
