diff --git a/debian/changelog b/debian/changelog
index a8facb9b..24fb6217 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+qemu-server (9.2.5+wsh1) trixie; urgency=medium
+
+  * Refresh patches: qemu-server,pve-qemu
+
+ -- hwinther <hwinther@users.noreply.github.com>  Sat, 08 Aug 2026 08:40:32 +0000
+
 qemu-server (9.2.5) trixie; urgency=medium
 
   * api: create disks: use correct volume name for a cloudinit disk on a
     snapshot-as-volume-chain storage
 
   * Improve cross-architecture support for query-machine-capabilities helper.
 
   * fix #7705: blockdev: detach backing chain on disk move/hot-unplug. When
     using snapshot-as-volume-chain, and doing snapshot, followed by disk move
     or hot-unplug, followed by removing the snapshot, all while the VM is
