diff --git a/debian/changelog b/debian/changelog
index f4b6e425..7fec17b5 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,22 @@
+qemu-server (9.1.5+wsh2) trixie; urgency=medium
+
+  * Manual fix needed: deb build failed (patch trigger) (#75)
+
+ -- Hans Christian Winther-Sorensen <hc@wsh.no>  Sat, 07 Mar 2026 12:09:31 +0000
+
+qemu-server (9.1.5+wsh1) trixie; urgency=medium
+
+  * feat: proxmox 9 patches (#56)
+
+ -- Hans Christian Winther-Sorensen <hc@wsh.no>  Fri, 06 Mar 2026 21:37:11 +0000
+
 qemu-server (9.1.5) trixie; urgency=medium
 
   * fix #1964: add 'guest-fsfreeze' QEMU guest agent setting to control
     whether filesystem freeze/thaw commands are issued for snapshots,
     replications and clones — not just backups. The previous
     'freeze-fs-on-backup' setting is deprecated in favor of the new one.
 
   * agent: file-read: allow specifying a byte offset and maximum byte count,
     and allow keeping the base64-encoding of the response content. This
     enables reading large files block-by-block and avoids inflated payload
