diff --git a/debian/changelog b/debian/changelog
index 1c6e989..af9ae76 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+qemu-server (8.2.5+wsh1) bookworm; urgency=medium
+
+  * Add remote audio and display support
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Mon, 11 Nov 2024 20:21:11 +0000
+
 qemu-server (8.2.5) bookworm; urgency=medium
 
   * remote migration: fix online migration via API clients
 
   * fix #5714: fix calloc parameter ordering
 
   * fix typos in user-visible strings
 
   * api: status: add some missing description for status return properties
 
