diff --git a/debian/changelog b/debian/changelog
index b8a627bd..c429fa92 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+qemu-server (9.1.17+wsh1) trixie; urgency=medium
+
+  * Manual fix needed: deb build failed (patch trigger) (#143)
+
+ -- Hans Christian Winther-Sorensen <hc@wsh.no>  Fri, 19 Jun 2026 03:38:42 +0000
+
 qemu-server (9.1.17) trixie; urgency=medium
 
   * cpu flags: include Hyper-V enlightenment flags in supported flags
 
   * partially fix #7094: when removing an external snapshot while the VM is
     running, ensure the snapshot-commit target is large enough, since it might
     still be smaller if the volume was resized since the snapshot was created
 
   * improve handling of local volumes during migration:
     - abort early when size or format cannot be determined
