diff --git a/debian/changelog b/debian/changelog
index 407c9df7..6fd95161 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+qemu-server (9.0.29+wsh1) trixie; urgency=medium
+
+  * feat: proxmox 9 patches (#56)
+
+ -- Hans Christian Winther-Sorensen <hc@wsh.no>  Tue, 18 Nov 2025 16:44:31 +0000
+
 qemu-server (9.0.29) trixie; urgency=medium
 
   * api: create vm: use ha-manager command to add VM as an HA resource, this
     fixes adding a VM to HA when using the qm CLI tool directly.
 
   * qmrestore CLI tool: allow starting a VM after it was restored successfully.
 
   * qmrestore CLI tool: allow adding a VM as an HA resource after it was
     restored.
 
