diff --git a/debian/changelog b/debian/changelog
index 27a457a9..3aba9aa5 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+qemu-server (9.0.30+wsh1) trixie; urgency=medium
+
+  * feat: proxmox 9 patches (#56)
+
+ -- Hans Christian Winther-Sorensen <hc@wsh.no>  Tue, 18 Nov 2025 22:29:08 +0000
+
 qemu-server (9.0.30) trixie; urgency=medium
 
   * api: add 'allow-ksm' to memory options that are allowed to be configured
     by non-root users.
 
   * vm start: ovmf: do not auto-enroll Microsoft UEFI CA 2023 for now, as this
     triggers an active Bitlocker inside the VM and the admin doing the restart
     of the VM might not have the Bitlocker recovery code available.
     Instead print a informational message in the start task log for now.
 
