diff --git a/debian/changelog b/debian/changelog
index 12c7d7cb..ead38be2 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+qemu-server (9.1.1+wsh1) trixie; urgency=medium
+
+  * feat: proxmox 9 patches (#56)
+
+ -- Hans Christian Winther-Sorensen <hc@wsh.no>  Wed, 10 Dec 2025 20:22:26 +0000
+
 qemu-server (9.1.1) trixie; urgency=medium
 
   * fix #7044: migration: cap the auto-increased allowed-downtime-limit to the
     maximum value that QEMU supports.
 
   * qmp client: bump timeout for synchronous block-dev snapshot commands to
     one hour, as they can take quite a bit of time, especially with qcow2
     internal snapshots.
 
   * qm cleanup: avoid potentially problematic cleanup during rollback.
