diff --git a/debian/changelog b/debian/changelog
index 6de0780d..292a4477 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+qemu-server (8.3.10+wsh1) bookworm; urgency=medium
+
+  * Add error message when arch binary is missing
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Mon, 07 Apr 2025 21:28:10 +0000
+
 qemu-server (8.3.10) bookworm; urgency=medium
 
   * drive: remove ancient and for a long time unsupported 'cow' from formats.
 
   * confidential virtualization: add AMD SEV-SNP support.
 
   * allow non-root users to add a virtion RNG (random number generator) device
     using /dev/u?random or /dev/hwrng as source.
 
   * config: add S3 and S4 power state properties to machine option.
