diff --git a/debian/changelog b/debian/changelog
index 15cc3937..9dbd1a09 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+qemu-server (9.2.10+wsh1) trixie; urgency=medium
+
+  * Refresh patches: qemu-server
+
+ -- hwinther <hwinther@users.noreply.github.com>  Sat, 10 Oct 2026 00:03:04 +0000
+
 qemu-server (9.2.10) trixie; urgency=medium
 
   * partially fix #7801: warn when configuring a CDROM drive with a VirtIO
     driver ISO version that is known to be problematic, or when starting a VM
     with such an ISO configured
 
   * fix #8030: snapshot: avoid orphaned vmstate volume on failure by moving
     volume allocation to the end of the vmstate saving helper
 
   * qemu-img: convert: cleanly separate options and arguments
