diff --git a/debian/changelog b/debian/changelog
index 49bda363..a13a0243 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+qemu-server (8.3.8+wsh1) bookworm; urgency=medium
+
+  * Add error message when arch binary is missing
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Fri, 04 Apr 2025 20:21:22 +0000
+
 qemu-server (8.3.8) bookworm; urgency=medium
 
   * resume: error out if VM is a template
 
   * backup: also restore VM power state for early failures
 
   * config: add special class that prevents writing the configuration
 
   * fix #6007: template backup: use minimized configuration for handling the full vm start
 
