diff --git a/debian/changelog b/debian/changelog
index ddb70dc..acb18de 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,3 +1,15 @@
+qemu-server (8.2.4+wsh2) bookworm; urgency=medium
+
+  * Update other docker files
+
+ -- Hans Christian Winther-Sorensen <hc@wsh.no>  Sun, 18 Aug 2024 20:29:28 +0000
+
+qemu-server (8.2.4+wsh1) bookworm; urgency=medium
+
+  * Improve changelog
+
+ -- Hans Christian Winther-Sorensen <hc@wsh.no>  Sun, 18 Aug 2024 20:25:22 +0000
+
 qemu-server (8.2.4) bookworm; urgency=medium
 
   * fix #5528: ensure cgroup limits like CPU weight are set correctly and in
