diff --git a/debian/changelog b/debian/changelog
index fa57e923..160d3c0a 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+qemu-server (9.2.4+wsh1) trixie; urgency=medium
+
+  * Refresh patches: qemu-server
+
+ -- hwinther <hwinther@users.noreply.github.com>  Fri, 07 Aug 2026 20:58:51 +0000
+
 qemu-server (9.2.4) trixie; urgency=medium
 
   * api: remote migrate tunnel: reject unknown parameters.
 
  -- Proxmox Support Team <support@proxmox.com>  Tue, 04 Aug 2026 13:38:58 +0200
 
 qemu-server (9.2.3) trixie; urgency=medium
 
   * cpu: default aarch64 guests to a KVM-usable CPU model
 
