diff --git a/debian/changelog b/debian/changelog
index 8949cf23..9617a508 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,16 @@
+qemu-server (9.1.0+wsh1) trixie; urgency=medium
+
+  * feat: proxmox 9 patches (#56)
+
+ -- Hans Christian Winther-Sorensen <hc@wsh.no>  Sun, 30 Nov 2025 20:21:51 +0000
+
 qemu-server (9.1.0) trixie; urgency=medium
 
   * vm start: check efi vars: add missing newline in print.
 
   * cpu config: nested-virt: update description of parameter to always
     recommend a CPU model similar to the host.
 
   * snapshot: prohibit live snapshot (remove) of qcow2 TPM drive on storage
     with snapshot-as-volume-chain for the time being.
 
