diff --git a/debian/changelog b/debian/changelog
index c531e7d5..045af3ce 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,10 +1,28 @@
+qemu-server (8.3.12+wsh3) bookworm; urgency=medium
+
+  * fix: Support for sun machine values
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Tue, 17 Jun 2025 00:10:52 +0000
+
+qemu-server (8.3.12+wsh2) bookworm; urgency=medium
+
+  * fix: Support for sun machine values
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Thu, 17 Apr 2025 13:45:18 +0000
+
+qemu-server (8.3.12+wsh1) bookworm; urgency=medium
+
+  * fix: Support for sun machine values
+
+ -- Hans Christian Winther-Sørensen <hc@wsh.no>  Thu, 17 Apr 2025 13:35:24 +0000
+
 qemu-server (8.3.12) bookworm; urgency=medium
 
   * virtiofs: drop unsafe and problematic writeback option completely fow now.
 
  -- Proxmox Support Team <support@proxmox.com>  Tue, 08 Apr 2025 17:30:49 +0200
 
 qemu-server (8.3.11) bookworm; urgency=medium
 
   * handle various deprecations of options in upstream QEMU 9.2.
 
